lib/dfe/analytics/entity_table_check_job.rb

```diff
@@ -25,18 +21,74 @@ def perform
         reschedule_job
       end
 
+      def build_event_for(model)
+        DfE::Analytics::Event.new
```

Comment:
@ericaporter @stevenleggdfe  - If the row_count is zero ie the table is empty, do we want to handle that situation ? Either not send the event at all OR make sure the code doesn't error both in dfe-analytics and in dataform.

Reply:
@stevenleggdfe what should I avoid for dataform?

Reply:
Good question @asatwal .

My preferred approach would be that if a table is empty we *do* send a checksum event to BigQuery with a ```row_count``` of ```0``` and a ```NULL``` value (not an empty ```value``` array - BQ can't store that - or an empty string as ```MD5("")``` produces a non null or empty value in GoogleSQL) for the checksum. Is that feasible?

Reply:
Note in Google SQL ```SELECT MD5("")``` produces ```1B2M2Y8AsgTpgAmY7PhCfg==``` !

Reply:
This might be important in cases where all the rows have been deleted in the database but for some reason we still have them in BQ because something went wrong with the deletion events.

Reply:
So for the Postgres SQL if run locally:

```SQL
SELECT count (*)  as row_count, MD5(STRING_AGG(CHECKSUM_TABLE.ID, '' ORDER BY CHECKSUM_TABLE.UPDATED_AT)) as checksum
FROM 
(SELECT trainees.id::TEXT as ID,
        updated_at as UPDATED_AT
 FROM trainees
 WHERE updated_at < '2020-11-05 09:25:19.863415'
 ORDER BY trainees.updated_at ASC) CHECKSUM_TABLE;
```
Result :
row_count  checksum
----------------------
0                  NULL

In Ruby:
Digest::MD5.hexdigest('')
=> "d41d8cd98f00b204e9800998ecf8427e"

In Postgres SQL:
``` SQL
SELECT MD5('');
```
Result:
md5
----------------------------------------
d41d8cd98f00b204e9800998ecf8427e

Which is different from BQ. Which makes me think do we have the same MD5 in Google SQL as Ruby and Postgres

Reply:
@asatwal good spot. It's not the same it appears.

However I think the difference is that we're missing some conversion to/from hexadecimal.

In GoogleSQL ```SELECT TO_HEX(MD5(""))``` produces ```d41d8cd98f00b204e9800998ecf8427e```, which matches.

I'll update the dfe-analytics-dataform logic to match.

Reply:
@stevenleggdfe I think sending nil values as an empty array is a design choice in dfe-analytics - even if I would try to send a nil value, it will get converted to an `[]`

Reply:
@ericaporter oh sorry, my bad! ```[]``` is fine. ```[NULL]``` is not. ```[""]``` is not great either.

Reply:
I have kept `MD5(COALESCE(STRING_AGG(CHECKSUM_TABLE.ID, '' ORDER BY CHECKSUM_TABLE.UPDATED_AT ASC), ''))` and `Digest::MD5.hexdigest(table_ids.join)` and both will return `[0, “d41d8cd98f00b204e9800998ecf8427e”]`